pkg/trace/api/container_linux.go

```diff
@@ -0,0 +1,157 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package api
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"os"
+	"regexp"
+	"sync"
+	"sync/atomic"
+	"syscall"
+	"time"
+)
+
+// cgroupPath is the path to the cgroup file where we can find the container id if one exists.
+var cgroupPath = "/proc/%d/cgroup"
+
+const (
+	uuidSource      = "[0-9a-f]{8}[-_][0-9a-f]{4}[-_][0-9a-f]{4}[-_][0-9a-f]{4}[-_][0-9a-f]{12}"
+	containerSource = "[0-9a-f]{64}"
+	taskSource      = "[0-9a-f]{32}-\\d+"
+)
+
+type ucredKey struct{}
```

Comment:
Please document everything. It helps a lot when new readers try to get familiar with this code. Of course one can figure it out anyway, but adding the extra effort once reduces that time multiple times in the future.